internal/pkg/v2/infrastructure/redis/device_profile.go

```diff
@@ -51,3 +51,39 @@ func addDeviceProfile(conn redis.Conn, dp model.DeviceProfile) (addedDeviceProfi
 
 	return dp, edgeXerr
 }
+
+// deviceProfileById query device profile by id from DB
+func deviceProfileById(conn redis.Conn, id string) (deviceProfile model.DeviceProfile, edgeXerr errors.EdgeX) {
+	obj, err := redis.Bytes(conn.Do(GET, id))
+	if err == redis.ErrNil {
+		return deviceProfile, errors.NewCommonEdgeX(errors.KindEntityDoesNotExist, "device profile doesn't exist in the database", err)
+	}
+	if err != nil {
+		return deviceProfile, errors.NewCommonEdgeX(errors.KindDatabaseError, "query device profile by id from the database failed", err)
+	}
+
+	err = json.Unmarshal(obj, &deviceProfile)
+	if err != nil {
+		return deviceProfile, errors.NewCommonEdgeX(errors.KindDatabaseError, "device profile format parsing failed from the database", err)
+	}
+
+	return
+}
+
+// deviceProfileByName query device profile by name from DB
+func deviceProfileByName(conn redis.Conn, name string) (deviceProfile model.DeviceProfile, edgeXerr errors.EdgeX) {
+	id, err := redis.String(conn.Do(HGET, DeviceProfileCollection+":name", name))
```

Comment:
write a general function into query.go for all the query by name, just like getObjectById

Reply:
Added the general function.